js/derivadex.js

```diff
@@ -0,0 +1,913 @@
+        //       },
+        //     ]
+        // }
+        return await this.parseTradesCustom (response, market, since, limit);
+    }
+
+    async getOrderIntents (trades) {
+        const result = {};
+        const params = {
+            'orderHash': [],
+        };
+        for (let i = 0; i < trades.length; i++) {
+            params['orderHash'].push (trades[i]['takerOrderHash']);
+        }
+        const orderIntentResponse = await this.publicGetOrderIntents (params);
+        const orderIntentResponseValue = orderIntentResponse['value'];
+        for (let i = 0; i < orderIntentResponseValue.length; i++) {
+            result[orderIntentResponseValue[i]['orderHash']] = orderIntentResponseValue[i];
+        }
+        return result;
+    }
+
+    async parseTradesCustom (trades, market = undefined, since = undefined, limit = undefined) {
+        trades = this.toArray (trades);
+        let result = [];
+        const orderIntents = await this.getOrderIntents (trades[0]);
+        for (let i = 0; i < trades[0].length; i++) {
+            const trade = await this.parseTradeCustom (trades[0][i], orderIntents);
+            result.push (trade);
+        }
+        result = this.sortBy2 (result, 'timestamp', 'id');
+        const symbol = (market !== undefined) ? market['symbol'] : undefined;
+        const tail = (since === undefined);
+        return this.filterBySymbolSinceLimit (result, symbol, since, limit, tail);
+    }
+
+    async parseTradeCustom (trade, orderIntents) {
+        const id = this.safeString (trade, 'takerOrderHash') + '_' + this.safeString (trade, 'epochId') + '_' + this.safeString (trade, 'txOrdinal');
+        const timestamp = this.parse8601 (this.safeString (trade, 'createdAt'));
+        const datetime = this.iso8601 (timestamp);
+        const symbol = this.safeString (trade, 'symbol');
+        const order = this.safeString (trade, 'takerOrderHash');
+        const price = this.safeString (trade, 'price');
+        const amount = this.safeString (trade, 'amount');
+        const fee = {
+            'cost': this.safeString (trade, 'takerFee'),
+            'currency': this.safeString (trade, 'takerFeeSymbol'),
+        };
+        const takerOrderHash = this.safeString (trade, 'takerOrderHash');
+        const sideNumber = this.safeInteger (orderIntents[takerOrderHash], 'side');
+        const orderTypeNumber = this.safeInteger (orderIntents[takerOrderHash], 'orderType');
+        const side = sideNumber === 0 ? 'buy' : 'sell';
+        let orderType = undefined;
+        if (orderTypeNumber === 0) {
+            orderType = 'limit';
+        } else if (orderTypeNumber === 1) {
+            orderType = 'market';
+        } else if (orderTypeNumber === 2) {
+            orderType = 'stop';
+        }
+        let takerOrMaker = 'taker';
+        if (trade['traderAddress'] !== undefined && trade['traderAddress'] !== this.safeString (trade['order_intents'][takerOrderHash], 'traderAddress')) {
```

Comment:
one thing to note is that liquidations wind up in the fills table (with reason = 1). You should inspect them, but you'll see that they have no takerOrderHash (only makerOrderHash)